### ISSUE TITLE
Deployment is not updating with new code

### ISSUE THREAD

**Issue by fabianboerner** [garden-io/garden#4093]
We want to deploy code in the frontend with garden but somehow garden doesn recognize the changes.

It always resolves that it is already deployed and not build the new version.

```
→ Getting build status for v-ceffa9f9c5... → Already built
ℹ campaign-frontend         → Deploying version v-70b90ea308...
✔ campaign-frontend         → Deploying version v-70b90ea308... → Already deployed
```
happens with buildkit

`garden deploy --force-refresh -l=verbose --logger-type=basic --var targetImage=production --env dev`

```
providers:
  - name: kubernetes
    namespace: ${var.namespace}
    environments: [dev, staging]
    buildMode: cluster-buildkit
    clusterBuildkit:
      cache:
        - type: registry
          registry:
            hostname: xxx
            namespace: xxx
    resources:
      builder:
        limits:
          memory: 8192
          cpu: 4000
    context: xxx
    defaultHostname: ${var.defaultHostname}
    setupIngressController: nginx
    deploymentRegistry:
      hostname: xxx
      namespace xxx
    imagePullSecrets:
      - name: xxx
        namespace: xxx
    tlsCertificates:
      - name: ${var.certificateSecretRefName}
        secretRef:
          name: ${var.certificateSecretRefName}
```

**Comment by vvagaytsev** [comment]
Thanks for reporting this @fabianboerner!
Could you share a bit more details and context, please?

* Which Garden version do you use?
* Please share the full project-level config and module config for `campaign-frontend`
* Please double-check your file exclusions in `dotIgnoreFiles` and `modules.exclude` in the project-level config, and `exclude` in the frontend module config

**Comment by fabianboerner** [comment]
Garden version 0.12.53

````
kind: Module

type: container

name: campaign-frontend

build:
  timeout: 4000
  targetImage: ${var.targetImage || "development"}
buildArgs:
  BASE_DOMAIN: ${var.base_domain}
  BASE_SUBDOMAIN: ${var.base_subdomain}
  APP_PORT: ${var.app_port}
  JOBS_API_ENDPOINT: ${var.jobs_api_endpoint}
  CAMPAIGNS_API_GRAPHQL_ENDPOINT: ${var.campaigns_api_graphql_endpoint}
  AUTH0_DOMAIN: ${var.auth0_domain}
  AUTH0_CLIENT_ID: ${var.auth0_client_id}
  AUTH0_AUDIENCE: ${var.auth0_audience}
  ROOT_FRONTEND_HOSTNAME: ${var.hostnames.root}
  STYLEGUIDE_FRONTNED_HOSTNAME: ${var.hostnames.styleguide}
  JOBS_FRONTEND_HOSTNAME: ${var.hostnames.jobs_frontend}
  MUI_LICENSE_KEY: ${local.env.MUI_LICENSE_KEY}
variables:
  $merge: ${var.commonEnvVars}
  app_port: 80
  campaigns_api_graphql_endpoint: xxx
  jobs_api_endpoint: xxx
  hostname: ${var.hostnames.campaign_frontend}
services:
  - name: campaign-frontend
    env:
      BASE_DOMAIN: ${var.base_domain}
      BASE_SUBDOMAIN: ${var.base_subdomain}
      APP_PORT: ${var.app_port}
      JOBS_API_ENDPOINT: ${var.jobs_api_endpoint}
      CAMPAIGNS_API_GRAPHQL_ENDPOINT: ${var.campaigns_api_graphql_endpoint}
      AUTH0_DOMAIN: ${var.auth0_domain}
      AUTH0_CLIENT_ID: ${var.auth0_client_id}
      AUTH0_AUDIENCE: ${var.auth0_audience}
      STYLEGUIDE_FRONTNED_HOSTNAME: ${var.hostnames.styleguide}
      JOBS_FRONTEND_HOSTNAME: ${var.hostnames.jobs_frontend}
      MUI_LICENSE_KEY: ${local.env.MUI_LICENSE_KEY}
    timeout: 4000
    ports:
      - name: http
        containerPort: ${var.app_port}
        servicePort: 80
    memory:
      min: 512
      max: 2046
    ingresses:
      - path: /
        hostname: ${var.hostname}
        port: http
    volumes:
      - name: campaign-frontend-volume
        containerPath: /usr/local/share/.cache/yarn
hotReload:
  sync:
    - target: /app

dockerfile: ${var.dockerfileName}
````

I have checked all ignore files and exclude settings

**Comment by vvagaytsev** [comment]
@fabianboerner is this still an issue in Garden Bonsai (0.13)? Please try it out with the latest 0.13 release. Thank you!

**Comment by vvagaytsev** [comment]
@fabianboerner is this still an issue in Garden Bonsai (0.13)? Please try it out with the latest 0.13 release. Thank you! Closing this as fixed in 0.13 for now. Feel free to reopen if it's still an issue.
